### PR TITLE
Switch browser tests to webpack

### PR DESCRIPTION
**Pull request by bergie**

### CODE REVIEW

**Comment by bergie** [comment]
@jonnor please review

File: spec/Component.coffee [hunk 1]
```diff
@@ -197,22 +193,22 @@ describe 'Component', ->
...
-      i = new socket.InternalSocket
+      i = new noflo.internalSocket.InternalSocket
       c.inPorts.in.attach(i)
       c.start()
       c.shutdown()
       chai.expect(c.started).to.equal(false)
       chai.expect(c.isStarted()).to.equal(false)
 
-  describe 'with object-based IPs', ->
+  describe 'with object-based noflo.IPs', ->
```
Comment (jonnor) [inline spec/Component.coffee]: Probably unintentional ;)

File: spec/Component.coffee [hunk 2]
```diff
@@ -197,22 +193,22 @@ describe 'Component', ->
...
 
-    it 'should speak IP objects', (done) ->
-      c = new component.Component
+    it 'should speak noflo.IP objects', (done) ->
```
Comment (jonnor) [inline spec/Component.coffee]: same

File: spec/Component.coffee [hunk 3]
```diff
@@ -317,16 +313,16 @@ describe 'Component', ->
...
-      sout2 = new socket.InternalSocket
+      sin1 = new noflo.internalSocket.InternalSocket
+      sin2 = new noflo.internalSocket.InternalSocket
+      sin3 = new noflo.internalSocket.InternalSocket
+      sout1 = new noflo.internalSocket.InternalSocket
+      sout2 = new noflo.internalSocket.InternalSocket
       done()
 
-    it 'should trigger on IPs', (done) ->
+    it 'should trigger on noflo.IPs', (done) ->
```
Comment (jonnor) [inline spec/Component.coffee]: unintentional

File: spec/Component.coffee [hunk 4]
```diff
@@ -411,12 +407,12 @@ describe 'Component', ->
...
-      sin1.post new IP 'data', 'AZ'
-      sin2.post new IP 'data', true
-      sin3.post new IP 'data', 'first'
+      sin1.post new noflo.IP 'data', 'AZ'
+      sin2.post new noflo.IP 'data', true
+      sin3.post new noflo.IP 'data', 'first'
 
-    it 'should receive and send complete IP objects', (done) ->
-      c = new component.Component
+    it 'should receive and send complete noflo.IP objects', (done) ->
```
Comment (jonnor) [inline spec/Component.coffee]: unintentional

File: spec/Component.coffee [hunk 5]
```diff
@@ -448,13 +444,13 @@ describe 'Component', ->
...
-      sin1.post new IP 'data', 'foo',
+      sin1.post new noflo.IP 'data', 'foo',
         groups: ['foo']
-      sin2.post new IP 'data', 'bar',
+      sin2.post new noflo.IP 'data', 'bar',
         groups: ['bar']
 
-    it 'should receive and send just IP data if wanted', (done) ->
-      c = new component.Component
+    it 'should receive and send just noflo.IP data if wanted', (done) ->
```
Comment (jonnor) [inline spec/Component.coffee]: unintentional

File: spec/Component.coffee [hunk 6]
```diff
@@ -480,14 +476,14 @@ describe 'Component', ->
...
 
-      sin1.post new IP 'data', 'foo',
+      sin1.post new noflo.IP 'data', 'foo',
         groups: ['foo']
-      sin2.post new IP 'data', 'bar',
+      sin2.post new noflo.IP 'data', 'bar',
         groups: ['bar']
 
-    it 'should receive IPs and be able to selectively find them', (done) ->
+    it 'should receive noflo.IPs and be able to selectively find them', (done) ->
```
Comment (jonnor) [inline spec/Component.coffee]: unintentional

File: spec/Component.coffee [hunk 7]
```diff
@@ -560,12 +556,12 @@ describe 'Component', ->
...
-      sin1.post new IP 'data', 'foo'
-      sin2.post new IP 'data', 'bar'
-      sin1.post new IP 'data', 'boo'
+      sin1.post new noflo.IP 'data', 'foo'
+      sin2.post new noflo.IP 'data', 'bar'
+      sin1.post new noflo.IP 'data', 'boo'
 
-    it 'should keep last data-typed IP packet for controls', (done) ->
-      c = new component.Component
+    it 'should keep last data-typed noflo.IP packet for controls', (done) ->
```
Comment (jonnor) [inline spec/Component.coffee]: unintentional

File: spec/ComponentLoader.coffee [hunk 8]
```diff
@@ -201,6 +199,8 @@ describe 'ComponentLoader with no external packages installed', ->
       chai.expect(instance.getIcon()).to.equal 'smile'
 
   describe 'reading sources', ->
+    before ->
+      return @skip() if noflo.isBrowser()
```
Comment (jonnor) [inline spec/ComponentLoader.coffee]: deserves comment why skipped

File: spec/ComponentLoader.coffee [hunk 9]
```diff
@@ -222,6 +222,8 @@ describe 'ComponentLoader with no external packages installed', ->
         done()
 
   describe 'writing sources', ->
+    before ->
+      return @skip() if noflo.isBrowser()
```
Comment (jonnor) [inline spec/ComponentLoader.coffee]: deserves comment why skipped

File: spec/Graph.coffee [hunk 10]
```diff
@@ -539,13 +539,13 @@ describe 'Graph', ->
         chai.expect(g.initializers.length).to.equal 0
 
   describe 'with no nodes', ->
-    g = new graph.Graph
+    g = new noflo.graph.Graph
     it 'should not allow adding edges', ->
       g.addEdge 'Foo', 'out', 'Bar', 'in'
-      chai.expect(graph.edges).to.be.empty
+      chai.expect(noflo.graph.edges).to.be.empty
```
Comment (jonnor) [inline spec/Graph.coffee]: Like you broke the test here? Or it was broken in first place? Should be fixed anyhow

File: spec/Graph.coffee [hunk 11]
```diff
@@ -539,13 +539,13 @@ describe 'Graph', ->
...
     it 'should not allow adding IIPs', ->
       g.addInitial 'Hello', 'Bar', 'in'
-      chai.expect(graph.initializers).to.be.empty
+      chai.expect(noflo.graph.initializers).to.be.empty
```
Comment (jonnor) [inline spec/Graph.coffee]: Same, think this check is broken

File: spec/Graph.coffee (hunk 10 above)
Comment (bergie) [inline spec/Graph.coffee]: Well spotted, this had always been wrong

**Comment by jonnor** [comment]
In general I think this kind of refactoring - where the changes in tests, should also work with the existing setup - should be done in a separate PR. Then the new things which actually change stuff, like going from component.io to webpack be in a next one (probably based on same).
But changes here are really minor, so guess it is fine.

**Comment by jonnor** [comment]
@bergie just making sure: getsource/setsource should still work for component.io builds, right?

**Comment by bergie** [comment]
@jonnor correct. And the test refractorings still work with Node which this PR doesn't change

**Pull request merged**